cmake/traccc-compiler-options-cpp.cmake

```diff
@@ -22,8 +22,10 @@ if( ( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" ) OR
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wextra" )
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wshadow" )
    traccc_add_flag( CMAKE_CXX_FLAGS "-Wunused-local-typedefs" )
-   traccc_add_flag( CMAKE_CXX_FLAGS "-pedantic" )
-   traccc_add_flag( CMAKE_CXX_FLAGS "-Wfloat-conversion" )
+   traccc_add_flag( CMAKE_CXX_FLAGS "-Wpedantic" )
+   if(PROJECT_IS_TOP_LEVEL)
+     traccc_add_flag( CMAKE_CXX_FLAGS "-Wfloat-conversion" )
```

Comment:
Curious. How does this flag cause warnings when building traccc as part of another project?

This line is only changing the compilation flags for the traccc source files themselves. It should have no effect on any client of the traccc headers. So, how does it produce warnings in Acts? :confused:

Reply:
traccc in ACTS includes ACTS headers, which then inherit traccc's compile flags. I'm working on making ACTS compile with `-Wfloat-conversion` but this will take a while.

Reply:
Ahh, I see. Yes, this back-and-forth dependence between the two projects makes things a bit hard to track...